services/tests/test_decoration.py

```diff
@@ -205,6 +205,44 @@ def test_decoration_type_limited_upload(self, enriched_pull, dbsession, mocker):
         assert decoration_details.decoration_type == Decoration.upload_limit
         assert decoration_details.reason == "Org has exceeded the upload limit"
 
+    def test_decoration_type_team_plan_upload_limit(
+        self, enriched_pull, dbsession, mocker
+    ):
+        mocker.patch("services.license.is_enterprise", return_value=False)
+        pr_author = OwnerFactory.create(
+            service="github",
+            username=enriched_pull.provider_pull["author"]["username"],
+            service_id=enriched_pull.provider_pull["author"]["id"],
+        )
+        dbsession.add(pr_author)
+        dbsession.flush()
+
+        enriched_pull.database_pull.repository.owner.plan = "users-teamm"
+        enriched_pull.database_pull.repository.private = True
+
+        commit = CommitFactory.create(
+            repository=enriched_pull.database_pull.repository,
+            author__service="github",
+            timestamp=datetime.now(),
+        )
+
+        report = ReportFactory.create(commit=commit)
+        for i in range(2499):
+            upload = UploadFactory.create(report=report, storage_path="url")
+            dbsession.add(upload)
+        dbsession.flush()
+
+        decoration_details = determine_decoration_details(enriched_pull)
+        assert decoration_details.decoration_type != Decoration.upload_limit
+        assert decoration_details.reason != "Org has exceeded the upload limit"
```

Comment:
Nice add to this test. I'll leave this optional, but would be nice to add this to the `test_decoration_type_limited_upload` test (and also rename that one to `test_decoration_type_basic_plan_limit`

Reply:
👍